utils/server-side-helper/app/cached-methods.ts

```diff
@@ -12,13 +15,54 @@ import {
 import { getUniteLegaleFromSlug } from '../../../models/core/unite-legale';
 
 const handleException = (e: any, slug: string) => {
-  if (e instanceof NotASirenError || e instanceof NotASiretError) {
-    logInfoInSentry(e);
-    redirect('/404');
-  }
-  if (e instanceof SirenNotFoundError || e instanceof SiretNotFoundError) {
-    logInfoInSentry(e);
-    redirect('/erreur/introuvable/' + slug);
+  try {
+    if (
+      e instanceof NotASirenError ||
+      e instanceof NotASiretError ||
+      e instanceof HttpNotFound
+    ) {
+      logWarningInSentry(
+        new Exception({
+          name: 'PageNotFoundException',
+          cause: e,
+          context: { slug },
+        })
+      );
+      redirect('/404');
+    } else if (
+      e instanceof SirenNotFoundError ||
+      e instanceof SiretNotFoundError
+    ) {
+      logWarningInSentry(
+        new Exception({
+          name: 'SirenNotFoundOrInvalid',
+          cause: e,
+          context: { slug },
+        })
+      );
+      redirect('/erreur/introuvable/' + slug);
+    } else if (e instanceof FetchRechercheEntrepriseException) {
+      logFatalErrorInSentry(e);
+      throw e;
+    } else {
+      logFatalErrorInSentry(
+        new Exception({
+          name: 'ServerErrorPageException',
+          cause: e,
+          context: { slug },
+        })
+      );
```

Comment:
I don't think this is necessary, since the error will be loggued anyway as Fatal in [`error.tsx`](
https://github.com/etalab/annuaire-entreprises-site/blob/6dbdaa8a10f6040e700b8cfad837d873f5e460c8/app/(header-home)/error.tsx#L10) page that will be displayed.

You can just throw the thing.